graylog-provider/src/test/java/com/busybusy/graylog_provider/GraylogProviderTest.java

```diff
@@ -0,0 +1,276 @@
+        provider.sendEvent(event);
+
+        lock.await(50L, TimeUnit.MILLISECONDS);
+
+        assertThat(logEventCalled).isTrue();
+        assertThat(testEventHashCode).isEqualTo(event.hashCode());
+        assertThat(loggedEventName).isEqualTo("Graylog Test Run No Params");
+    }
+
+    @Test
+    public void testSendEvent_unTimed_withParams() throws InterruptedException
+    {
+        AnalyticsEvent event = new AnalyticsEvent("Graylog Event With Params Run")
+                .putAttribute("some_param", "yes")
+                .putAttribute("another_param", "yes again");
+
+        provider.sendEvent(event);
+
+        lock.await(50L, TimeUnit.MILLISECONDS);
+
+        assertThat(loggedEventName).isEqualTo("Graylog Event With Params Run");
+        assertThat(logEventCalled).isTrue();
+        assertThat(testEventHashCode).isEqualTo(event.hashCode());
+    }
+
+    @Test
+    public void testLogContentViewEvent() throws InterruptedException
+    {
+        ContentViewEvent event = new ContentViewEvent("Test page 7");
+        provider.sendEvent(event);
+
+        lock.await(50L, TimeUnit.MILLISECONDS);
+
+        assertThat(loggedEventName).isEqualTo(CommonEvents.CONTENT_VIEW);
+        assertThat(logEventCalled).isTrue();
+        assertThat(testEventHashCode).isEqualTo(event.hashCode());
+    }
+
+    @Test
+    public void testLogErrorEvent() throws InterruptedException
+    {
+        ErrorEvent event = new ErrorEvent()
+                .setMessage("something bad happened")
+                .setException(new EmptyStackException());
+        provider.sendEvent(event);
+
+        lock.await(50L, TimeUnit.MILLISECONDS);
+
+        assertThat(loggedEventName).isEqualTo(CommonEvents.ERROR);
+        assertThat(logEventCalled).isTrue();
+        assertThat(testEventHashCode).isEqualTo(event.hashCode());
+    }
+
+    @Test
+    public void testSendEvent_timed_noParams()
+    {
+        AnalyticsEvent event = new AnalyticsEvent("Graylog Timed Event")
+                .setTimed(true);
+
+        provider.sendEvent(event);
+        assertFalse(logEventCalled);
+    }
+
+    @Test
+    public void testSendEvent_timed_withParams()
+    {
+        AnalyticsEvent event = new AnalyticsEvent("Graylog Timed Event With Parameters")
+                .setTimed(true)
+                .putAttribute("some_param", "yes")
+                .putAttribute("another_param", "yes again");
+
+        provider.sendEvent(event);
+        assertFalse(logEventCalled);
+    }
+
+    @Test
+    public void testEndTimedEvent_Valid() throws InterruptedException
+    {
+        AnalyticsEvent event = new AnalyticsEvent("Graylog Timed Event With Parameters")
+                .setTimed(true)
+                .putAttribute("some_param", "yes")
+                .putAttribute("another_param", "yes again");
+
+        provider.sendEvent(event);
+        assertFalse(logEventCalled);
+
+        try
+        {
+            Thread.sleep(50);
```

Comment:
Nope. At one point I was validating that we get a positive duration on the timed event, but now that the event gets serialized I don't care about parsing the JSON I would get out of the callback.